src/algorithm.jl

```diff
@@ -320,8 +320,15 @@ function solve_subproblem(model::Optimizer, mip_solution, comp::Function)
     if (model.nlp_block !== nothing && !isempty(model.nlp_block.constraint_bounds) && model.nl_slack_variables === nothing) ||
         (!isempty(model.quad_less_than) && model.quad_less_than_slack_variables === nothing) ||
         (!isempty(model.quad_greater_than) && model.quad_greater_than_slack_variables === nothing)
-        obj = MOI.ScalarAffineFunction([MOI.ScalarAffineTerm(1.0, x) for x in model.nl_slack_variables], 0.0)
+
+        if model.nl_slack_variables !== nothing
+            obj = MOI.ScalarAffineFunction([MOI.ScalarAffineTerm(1.0, x) for x in model.nl_slack_variables], 0.0)
+        else
+            obj = MOI.ScalarAffineFunction(Array{MOI.ScalarAffineTerm{Float64}}([]), 0.0)
```

Comment:
Use `MOI.ScalarAffineTerm[]` instead of creating a vector of `Any` and then convert it